File: bsi-backend/app/models/signal.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, Text, JSON, Enum
from app.db.base import Base
from app.enums.signal import SignalStatus

class Signal(Base):
    __tablename__ = "signals"

    id = Column(Integer, primary_key=True, index=True)
    raw_log_id = Column(Integer, ForeignKey("raw_webhook_logs.id"), nullable=False)
    payload_version = Column(String, nullable=False)
    external_signal_id = Column(String, unique=True, nullable=False, index=True)
    domain = Column(String, nullable=False, default="CRYPTO_SPOT")
    asset = Column(String, nullable=False)
    market = Column(String, nullable=False)
    timeframe = Column(String, nullable=False)
    signal_type = Column(String, nullable=False)
    signal_direction = Column(String, nullable=False)
    trigger_price = Column(Float, nullable=False)
    signal_timestamp = Column(DateTime, nullable=False)
    strategy_name = Column(String, nullable=False)
    strategy_version = Column(String, nullable=False)
    indicator_snapshot_json = Column(JSON, nullable=False)
    candle_context_json = Column(JSON, nullable=False)
    status = Column(Enum(SignalStatus), nullable=False, default=SignalStatus.RECEIVED)
    status_message = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

File: frontend/src/types/signal.ts
export type SignalStatus = 
  | 'RECEIVED' | 'VALIDATED' | 'STORED' | 'PENDING_INTERPRETATION'
  | 'INTERPRETED' | 'INTERPRETATION_FAILED' | 'REVIEWED' | 'CLOSED' | 'ERROR';

export type InterpretationStatus = 'PENDING' | 'PROCESSING' | 'SUCCESS' | 'FAILED';
export type OutcomeStatus = 'WIN' | 'LOSS' | 'BREAKEVEN' | 'OPEN' | 'EXPIRED' | 'INCONCLUSIVE';

export interface PerformanceSummary {
  total_signals: number;
  n_resolved: number;
  resolved_rate: number;
  open_rate: number;
  interpreted_success_rate: number;
  interpreted_failed_rate: number;
  no_interpretation_rate: number;
  win_rate_global: number;
  win_rate_ia_success: number;
  n_ia_success: number;
  avg_return_pct: number;
  median_return_pct: number;
  data_quality_label: string;
  domain?: string;
}

export interface LLMImpactAnalysis {
  group: string;
  n: number;
  win_rate: number;
  avg_return_pct: number;
  median_return_pct: number;
  resolved_rate: number;
}

export interface DimensionBreakdown {
  label: string;
  n: number;
  win_rate: number;
  avg_return_pct: number;
  median_return_pct: number;
  sample_too_small: boolean;
}

export interface SignalListItem {
  id: number;
  domain?: string;
  external_signal_id: string;
  strategy_name: string;
  asset: string;
  timeframe: string;
  signal_direction: 'LONG' | 'SHORT';
  trigger_price: number;
  status: SignalStatus;
  created_at: string;
  interpretation?: {
    status: InterpretationStatus;
    regime: string;
    confidence_level: 'HIGH' | 'MEDIUM' | 'LOW';
    confidence_score: number;
    context_alignment: string;
  };
  outcome?: {
    status: OutcomeStatus;
    return_pct: number;
  };
}

export interface SignalDetail extends Omit<SignalListItem, 'interpretation' | 'outcome'> {
  market: string;
  signal_type: string;
  strategy_version: string;
  indicator_snapshot_json: any;
  candle_context_json: any;
  updated_at: string;
  interpretation_full?: any;
  outcome_full?: any;
}

export interface SystemStatus {
  app_name: string;
  environment_mode: string;
  database_ok: boolean;
  counts: {
    signals: number;
    interpretations: number;
    outcomes: number;
  };
  indicators: {
    seeded_data_present: boolean;
    real_captured_present: boolean;
  };
}
